cmake/modules/packages/FindHDFS.cmake

```diff
@@ -24,23 +24,38 @@ The following vars are set if HDFS (the native library provided with Hadoop) is
 
 include(FindPackageHandleStandardArgs)
 
+find_library(JAVA_JVM_LIBRARY NAMES jvm PATH_SUFFIXES native)
```

Comment:
it seems to me that you should rather use https://cmake.org/cmake/help/latest/module/FindJNI.html and
do a find_package(JNI) to get this variable